### ISSUE TITLE
Relative port id X is not unique

### ISSUE THREAD

**Issue by ishioni** [SCST-project/scst#230]
I am using scst in truenas scale, and using iscsi as a backing store for a kubernetes provisioner. If enough operations happen at the same time, it seems that scst becomes confused and this error pops up. Clients will never be able to connect, and only restarting scst fixes the problem (killing all the connections however)

Truenas SCALE version: 23.10.2
uname: Linux quadrant 6.1.74-production+truenas #2 SMP PREEMPT_DYNAMIC Wed Feb 21 20:30:38 UTC 2024 x86_64 GNU/Linux
scst version: 3.8.0.17844fb2f-~truenas+2

```
Apr 17 21:37:18 quadrant iscsi-scstd[2218239]: Connect from 10.1.2.40:55744 to 10.1.2.2:3260
Apr 17 21:37:18 quadrant kernel: [2218239]: scst: Using security group "security_group" for initiator "iqn.2016-04.com.open-iscsi:756c997f796d#10.1.2.2" (target iqn.2011-08.org.truenas.ctl:pvc-3bce22af-93b7-4b2b-b430-c124bdb16b81)
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst: Session 000000001b0a6fac created: target 000000005e4cf4df, tid 5, sid 0x190100003d0200, initiator iqn.2016-04.com.open-iscsi:756c997f796d#10.1.2.2
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst: Negotiated parameters: InitialR2T No, ImmediateData Yes, MaxConnections 1, MaxRecvDataSegmentLength 1048576, MaxXmitDataSegmentLength 262144, 
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst:     MaxBurstLength 1048576, FirstBurstLength 65536, DefaultTime2Wait 0, DefaultTime2Retain 0, 
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst:     MaxOutstandingR2T 1, DataPDUInOrder Yes, DataSequenceInOrder Yes, ErrorRecoveryLevel 0, 
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst:     HeaderDigest None, DataDigest None, OFMarker No, IFMarker No, OFMarkInt 2048, IFMarkInt 2048, RDMAExtensions No
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst: Target parameters set for session 190100003d0200: QueuedCommands 32, Response timeout 90, Nop-In interval 30, Nop-In timeout 30
Apr 17 21:37:18 quadrant kernel: [2218239]: iscsi-scst: Creating connection 000000001228d043 for sid 0x190100003d0200, cid 0 (initiator iqn.2016-04.com.open-iscsi:756c997f796d#10.1.2.2)
Apr 17 21:37:42 quadrant kernel: [2218180]: scst: ***ERROR***: Relative port id 3 is not unique
Apr 17 21:37:44 quadrant kernel: [1564238]: dev_vdisk: Registering virtual vdisk_blockio device pvc-6bda841d-798b-401a-b884-5a6da7e46307 (BLOCKIO)
Apr 17 21:37:44 quadrant kernel: [1564238]: dev_vdisk: Auto enable thin provisioning for device /dev/zvol/SSD/talos-applications/pvc-6bda841d-798b-401a-b884-5a6da7e46307
Apr 17 21:37:44 quadrant kernel: [1564238]: dev_vdisk: Attached SCSI target virtual disk pvc-6bda841d-798b-401a-b884-5a6da7e46307 (file="/dev/zvol/SSD/talos-applications/pvc-6bda841d-798b-401a-b884-5a6da7e46307", fs=20480MB, bs=4096, nblocks=5242880, cyln=2560)
Apr 17 21:37:44 quadrant kernel: [1564238]: scst: Added device pvc-6bda841d-798b-401a-b884-5a6da7e46307 to group copy_manager_tgt (LUN 19, flags 0x4) to target copy_manager_tgt
Apr 17 21:37:44 quadrant kernel: [1564238]: scst: Attached to virtual device pvc-6bda841d-798b-401a-b884-5a6da7e46307 (id 18)
Apr 17 21:37:44 quadrant kernel: [1564208]: dev_vdisk: USN for device pvc-6bda841d-798b-401a-b884-5a6da7e46307 changed to 739bfdef5a20ee2
Apr 17 21:37:44 quadrant kernel: [2218180]: scst: ***ERROR***: Relative port id 3 is not unique
Apr 17 21:38:02 quadrant iscsi-scstd[2218239]: Connect from 10.1.2.41:49608 to 10.1.2.2:3260
Apr 17 21:38:04 quadrant iscsi-scstd[2218239]: Connect from 10.1.2.41:49620 to 10.1.2.2:3260
Apr 17 21:38:06 quadrant iscsi-scstd[2218239]: Connect from 10.1.2.41:33416 to 10.1.2.2:3260
```

**Comment by bmeagherix** [comment]
@ishioni I believe this was a regression (NAS-128099) that was introduced in SCALE, rather than being a problem wrt scst per-se.  It has been fixed in the latest TrueNAS SCALE 24.04.0 which was released a couple of days ago.

**Comment by ishioni** [comment]
Indeed it has. Sorry for the noise and feel free to close
